ci/lint/check_cpp_files_inclusion.py

```diff
@@ -38,18 +14,19 @@ def check_ray_core_inclusion(fname: str):
         ImportError: if incorrect inclusion is detected.
     """
     # Exclude protobuf, which requires absolution path for compilation.
-    cmd = f"grep 'src/ray' {fname} | grep -v 'pb'"
-    grep_res = subprocess.check_output(cmd, shell=True, text=True)
-    if grep_res:
-        raise ImportError(f"{fname} has invalid header file inclusion: {grep_res}")
+    pattern = re.compile(r"^(?=.*src/ray)(?!.*pb)")
+
+    with open(fname, "r") as file:
+        for line in file:
+            if pattern.search(line):
+                raise ImportError(f"{fname} has invalid header file inclusion: {line}")
```

Comment:
I suggest you to raise user-defined error here. [ImportError](https://docs.python.org/3.12/library/exceptions.html#ImportError) is used when the `import` statement has troubles, so raise `ImportError` here is confusing.

You can define a custom error like this:

```python
class InvalidCppHeaderInclusion(Exception):
    pass
```

And then `raise InvalidCppHeaderInclusion`